subcommands/keys/ca_add_device_ca.go

```diff
@@ -0,0 +1,109 @@
+
+This command can add one or both of the following certificates:
+
+### online-ca - A Foundries.io owned keypair to support lmp-device-register.
+In order for lmp-device-register to work, Foundries.io needs the ability to sign client certificates for devices.
+If enabled, the factory_ca keypair will sign the certificate signing request returned from the API.
+If the online-ca was already created earlier, a new onlin-ca will replace it for the registration process.
+Still, the previous online-ca will be present in a list of the device CAs trusted by the device gateway,
+so that devices with client certificates issues by it may continue to connect to Foundries.io services.
+
+### local-ca - A keypair you own that can be used for things like your manufacturing process,
+where you may generate device client certificates without having to communicate with Foundries.io web services.
+You can create as many local-ca files as you need, and use each of them to generate device client certificates.
+All such CAs will be added to the list of the device CAs trusted by the device gateway.`,
+	}
+	caCmd.AddCommand(cmd)
+	cmd.Flags().BoolP("online-ca", "", false,
+		"Create an online CA owned by Foundries that works with lmp-device-register")
+	cmd.Flags().BoolP("local-ca", "", false,
+		"Create a local CA that you can use for signing your own device certificates")
+	cmd.Flags().StringP("local-ca-filename", "", x509.DeviceCaCertFile,
+		fmt.Sprintf("A file name of the local CA (only needed if the %s file already exists)", x509.DeviceCaCertFile))
+	_ = cmd.MarkFlagFilename("local-ca-filename")
+	// HSM variables defined in ca_create.go
+	cmd.Flags().StringVarP(&hsmModule, "hsm-module", "", "", "Load a root CA key from a PKCS#11 compatible HSM using this module")
+	cmd.Flags().StringVarP(&hsmPin, "hsm-pin", "", "", "The PKCS#11 PIN to log into the HSM")
+	cmd.Flags().StringVarP(&hsmTokenLabel, "hsm-token-label", "", "", "The label of the HSM token containing the root CA key")
+}
+
+func doAddDeviceCa(cmd *cobra.Command, args []string) {
+	factory := viper.GetString("factory")
+	createLocalCA, _ := cmd.Flags().GetBool("local-ca")
+	createOnlineCA, _ := cmd.Flags().GetBool("online-ca")
+	localCaFilename, _ := cmd.Flags().GetString("local-ca-filename")
+
+	if !createLocalCA && !createOnlineCA {
+		subcommands.DieNotNil(errors.New("At least one of --online-ca or --local-ca must be true"))
+	}
+
+	subcommands.DieNotNil(os.Chdir(args[0]))
+	hsm, err := x509.ValidateHsmArgs(
+		hsmModule, hsmPin, hsmTokenLabel, "--hsm-module", "--hsm-pin", "--hsm-token-label")
+	subcommands.DieNotNil(err)
+	x509.InitHsm(hsm)
+
+	fmt.Println("Fetching a list of existing device CAs")
+	resp, err := api.FactoryGetCA(factory)
+	subcommands.DieNotNil(err)
+	certs := client.CaCerts{CaCrt: resp.CaCrt}
+	if len(certs.CaCrt) == 0 {
+		subcommands.DieNotNil(errors.New("Please, initialize your factory PKI using 'fioctl keys ca create'."))
```

Comment:
I liked your comment about Error + "Please" on a previous PR. Something like that may be more appropriate as an error message:
```suggestion
		subcommands.DieNotNil(errors.New("Factory PKI not initialized. Set it up using 'fioctl keys ca create'."))
```

Reply:
We may keep the "Please" after "Factory PKI not initialized.", but I personally like it better without it.